design/specs/column-extraction.md

```diff
@@ -8,87 +8,66 @@ editor: markdown
 
-    The user chooses all the columns that will be attributes for the new object. For example, in the instance of the publisher, there could be columns for 'Publisher Name' and 'Publisher Id.' Including all the attributes is very important since columns cannot be moved after the fact.
+![image](/assets/design/specs/column-extraction/185139636-9a8a048c-9f87-4e26-aa2c-88f4e8d1648d.png)
 
-     <img width="1512" alt="image" src="https://user-images.githubusercontent.com/845767/177544077-9f074830-ce64-4ed9-80bc-53de53977f9c.png">
+## Extract Columns Dialog
 
+The dialog will list the columns that will be extracted and the table that will be created. The user can change the name of the new table. The dialog will also list the links that will be created between the new table and the original table. The user can change the name of the link column.
 
-3. **The user clicks on the 'Extract Columns' option**
+![image](/assets/design/specs/column-extraction/185901982-f991a6a3-c228-4c94-9d57-55b921e44511.png)
 
-    The user can choose to keep copies of the columns in the table if needed. By default the extracted columns are deleted.
+### Impact of the Extract Columns Operation
 
-4. **The user inspects the new foreign key column (link)**
+Since the split operation will affect the original table, the user should be made aware of the operation's impact. The dialog should communicate the following:
 
-    The new foreign key column will contain links to the new records that have been created. Additionally, when selected, the inspector panel will display the link properties.
+- The original table will be updated to remove the extracted columns
+- A new table will be created with the extracted columns
+- A link will be created between the new table and the original table
 
-    <img width="1469" alt="image" src="https://user-images.githubusercontent.com/845767/177546415-23b11ded-8987-4e69-ab1a-1decb9e773d3.png">
+### Link Column Naming Convention
 
+It was decided that the link column naming convention should be as follows:
 
-5. **The user navigates to the new table**
+>In the application, where singularization is not straightforward, we don’t suggest column names. Instead we require the user to manually name the FK column.
 
-    The new table will be listed in the list of tables for the current schema. The user can modify the table, add new columns or rename it if needed.
+In this case, since we cannot infer the new table name from the selected columns, then the user must manually enter both, new table name and link column name.
 
-    <img width="874" alt="image" src="https://user-images.githubusercontent.com/845767/177550009-30fcae81-2ae9-4539-ad85-5a0d1289790b.png">
+#### Using Rules to Generate Column Names
```

Comment:
We would be generating a single foreign key column, so shouldn't it be `Column Name` instead of `Column Names`?